src/features/singleRight/delegate/ChooseRightsPage/ChooseRightsPage.tsx

```diff
@@ -1,17 +1,239 @@
+import { DualElementsContainer, Page, PageContainer, PageContent, PageHeader } from '@/components';
+import { useAppDispatch, useAppSelector } from '@/rtk/app/hooks';
+import { SingleRightPath } from '@/routes/paths';
+import { useMediaQuery } from '@/resources/hooks';
+import {
+  type ChosenService,
+  removeServiceResource,
+} from '@/rtk/features/singleRights/singleRightsSlice';
+import { getSingleRightsErrorCodeTextKey } from '@/resources/utils/errorCodeUtils';
+
+import { ResourceActionBar } from '../ChooseServicePage/ResourceActionBar/ResourceActionBar';
+
+import classes from './ChooseRightsPage.module.css';
+
+interface DelegationResourceDTO {
+  title: string | undefined;
+  serviceIdentifier: string | undefined;
+  rightKey: string;
+  checked: boolean;
+}
 
 export const ChooseRightsPage = () => {
   const { t } = useTranslation();
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const [popoverOpen, setPopoverOpen] = useState(false);
+  const [checkedStates, setCheckedStates] = useState<Record<string, boolean>>({});
+  const isSm = useMediaQuery('(max-width: 768px)');
+  const rightsToBeDelegated: DelegationResourceDTO[] = [];
+
+  const chosenServices = useAppSelector((state) => state.singleRightsSlice.chosenServices);
+
+  const delegableChosenServices = chosenServices.filter((s) => s.status !== 'NotDelegable');
+
+  const onRemove = (identifier: string | undefined) => {
+    void dispatch(removeServiceResource(identifier));
+  };
+
+  const onConfirm = () => {
+    setPopoverOpen(!popoverOpen);
+    const filteredList = rightsToBeDelegated.filter((right) => right.checked);
+    const listWithoutChecked = filteredList.map(({ checked, ...rest }) => rest);
+  };
+
+  const handleToggleChecked = (serviceIdentifier: string, rightKey: string) => {
+    setCheckedStates((prevCheckedStates) => ({
+      ...prevCheckedStates,
+      [`${serviceIdentifier}-${rightKey}`]: !prevCheckedStates[`${serviceIdentifier}-${rightKey}`],
+    }));
+  };
+
+  const sortedServiceResources = useMemo(() => {
+    return [...delegableChosenServices]?.sort((a, b) => {
+      const isPartiallyDelegableA = a.status === 'PartiallyDelegable';
+      const isPartiallyDelegableB = b.status === 'PartiallyDelegable';
+
+      if (isPartiallyDelegableA && !isPartiallyDelegableB) {
+        return -1;
+      }
+      if (!isPartiallyDelegableA && isPartiallyDelegableB) {
+        return 1;
+      }
+
+      return a.service?.title.localeCompare(b.service.title);
+    });
+  }, [delegableChosenServices]);
+
+  const serviceResourcesActionBars = useMemo(() => {
+    return sortedServiceResources?.map((chosenService: ChosenService) => {
+      let hasPartiallyDelegableAppeared = false;
+      const isPartiallyDelegable =
+        chosenService.status === 'PartiallyDelegable' && !hasPartiallyDelegableAppeared;
+
+      if (isPartiallyDelegable) {
+        hasPartiallyDelegableAppeared = true;
```

Comment:
It's being used to only open the first ActionBar. Although I see it has no function as it is right now. Because I cleant up the code and thought I could move it to become a local variable as it is now, I see it breaks the function and I'll remove it outside. Then it will work